plugins/woocommerce-admin/client/marketplace/contexts/product-list-context.tsx

```diff
@@ -0,0 +1,99 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect, createContext } from '@wordpress/element';
+
+type ProductListItemType = {
+	title: string;
+	image: string;
+	excerpt: string;
+	link: string;
+	demo_url: string;
+	price: string;
+	hash: string;
+	slug: string;
+	id: number;
+	rating: number | null;
+	reviews_count: number | null;
+	vendor_name: string;
+	vendor_url: string;
+	icon: string;
+};
+
+type ProductListContextType = {
+	productList: ProductListItemType[];
+	setSearchTerm: ( searchTerm: string ) => void;
+	setCategory: ( category: string ) => void;
+	isLoading: boolean;
+};
+
+export const ProductListContext = createContext< ProductListContextType >( {
+	productList: [],
+	setSearchTerm: () => {},
+	setCategory: () => {},
+	isLoading: false,
+} );
+
+type ProductListContextProviderProps = {
+	children: JSX.Element;
+	country?: string;
+	locale?: string;
+};
```

Comment:
This feels a bit repetitive. I wonder if there is a better way to make TypeScript infer the type from the default values I pass to `createContext`.

Reply:
I'm trying to figure out if an interface would help here 🤔